File: SchoolWebApp-Client/src/app/academics/models/grade.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import {Curriculum} from './curriculum';

export class Grade extends ResourceModel<Grade> {
    public name?: string;
    public abbr?: string;

    public minScore?: number;
    public maxScore?: number;
    public points?: number;

    public remarksSwa?: string;
    public remarksEng?: string;

    public curriculumId?: string;
    public curriculum?: Curriculum;

    public rank?: number;
    public category?: string;

    constructor(model?: Partial<Grade>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/Academics/Grade.cs
﻿using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class Grade : Base
    {
        [Required(ErrorMessage = "Enter the grade name")]
        [Display(Name = "Grade name")]
        [StringLength(255)]
        public required string Name { get; set; }

        [Required(ErrorMessage = "Enter the abbreviation")]
        [Display(Name = "Grade abbreviation")]
        [StringLength(255)]
        public required string Abbr { get; set; }

        [Required(ErrorMessage = "Enter the minimum score")]
        [Display(Name = "Minimum score")]
        public float MinScore { get; set; }

        [Required(ErrorMessage = "Enter the maximum score")]
        [Display(Name = "Maximum score")]
        public float MaxScore { get; set; }

        [Required(ErrorMessage = "Enter the grade points")]
        public float Points { get; set; }

        [Display(Name = "Remarks in Kiswahili")]
        public string? RemarksSwa { get; set; }

        [Display(Name = "Remarks in English")]
        public string? RemarksEng { get; set; }
        public int Rank { get; set; }
        public int CurriculumId { get; set; }
        public Curriculum? Curriculum { get; set; }
    }
}
